demonstrations/tutorial_qutrits_bernstein_vazirani.py

```diff
@@ -0,0 +1,394 @@
+
+I invite you to take your time to think of a possible strategy (at the classical level) in order to determine :math:`\vec{a}` with the minimum number of evaluations of the function :math:`f`. The optimal solution requires only :math:`n` calls to the function! Let's see how we can do this.
+Knowing the form of :math:`\vec{a}` and :math:`\vec{x}`, we can rewrite :math:`f` as:
+
+.. math::
+  f(\vec{x})=\sum_{i=0}^{n-1}a_ix_i \pmod 2.
+
+The strategy will be to deduce one element of :math:`\vec{a}` with each call to the function. To do this, imagine that we want to determine the value :math:`a_i`. If we notice, we will simply have to choose :math:`\vec{x}` as a vector of all zeros except a one in the i-th position, since in this case:
+
+
+.. math::
+    f(\vec{x})= 0\cdot a_0 + 0\cdot a_1 + ... + 1\cdot a_i + ... + 0\cdot a_{n-1} \pmod 2 \quad= a_i.
+
+It is trivial to see therefore that :math:`n` questions are needed. The question is: can we do it more efficiently with a quantum computer? The answer is yes, and in fact, we only need to make one call to the function! The first step is to see how we can represent this statement in a circuit. In this case, we will assume that we have an oracle :math:`U_f` that encodes the function, as we can see in the picture.
+
+.. figure:: ../demonstrations/qutrits_bernstein_vazirani/oracle_qutrit.jpg
+   :scale: 35%
+   :alt: Oracle representation
+   :align: center
+
+   Oracle definition.
+
+In general, :math:`U_f` sends the basic state :math:`|\vec{x} \rangle |y\rangle` into the state :math:`| \vec{x} \rangle |y + \vec{a} \cdot \vec{x} \pmod{2} \rangle`.
+Suppose, for example, that :math:`\vec{a}=[0,1,0]`. Then :math:`U_f|1110\rangle = |1111\rangle`, since we are evaluating :math:`f` at the point :math:`\vec{x} = [1,1,1]`. The scalar product between the two values is :math:`1`, so the last qubit of the output will take the value :math:`1`. That said, the Bernstein-Vazirani algorithm states the following:
+
+.. figure:: ../demonstrations/qutrits_bernstein_vazirani/bernstein_vazirani_algorithm.jpg
+   :scale: 35%
+   :alt: Bernstein-Vazirani's algorithm
+   :align: center
+
+   Bernstein Vazirani algorithm.
+
+
+What we can see is that by simply using Hadamard gates before and after the oracle, after a single run the output of the circuit is exactly the hidden value of :math:`\vec{a}`. Let's do a little math to verify that this is so.
+
+- First, the input to our circuit is :math:`|0001\rangle`.
+
+- The second step is to apply Hadamard gates on this state, and for this we must use the following property
+
+  .. math::
+        H^n|\vec{x}\rangle = \frac{1}{\sqrt{2^n}}\sum_{\vec{z} \in \{0,1\}^n}(-1)^{\vec{x}\cdot\vec{z}}|\vec{z}\rangle.
+
+  This expression may seem complicated to obtain but understanding the case of one qubit, we would have to multiply and group. It is simply to realize that :math:`H|x\rangle = \frac{|0\rangle+(-1)^x |1\rangle}{\sqrt{2}}` where :math:`x \in \{0.1\}`.

```

Comment:
```suggestion
  This expression may seem complicated but understanding the case of one qubit, we would have to multiply and group. It is simply to realize that :math:`H|x\rangle = \frac{|0\rangle+(-1)^x |1\rangle}{\sqrt{2}}` where :math:`x \in \{0, 1\}`.

```

In the last ket, maybe use a different letter than x, just to distinguish between the bit string \vec{x} and a single bit x. 

Also, the expression "the case of one qubit, we would have to multiply and group" is a little unclear.